standardization/src/main/scala/za/co/absa/enceladus/standardization/StandardizationJob.scala

```diff
@@ -171,13 +172,17 @@ object StandardizationJob {
     }
   }
 
-  private def getCsvOptions(cmd: CmdConfig): HashMap[String,Option[RawFormatParameter]] = {
+  private def getCsvOptions(cmd: CmdConfig, schema: Option[StructType] = None): HashMap[String,Option[RawFormatParameter]] = {
     if (cmd.rawFormat.equalsIgnoreCase("csv")) {
+      val knownNumberOfFields = if (schema.isDefined) schema.get.fields.length else 0
       HashMap(
         "delimiter" -> cmd.csvDelimiter.map(StringParameter),
         "header" -> cmd.csvHeader.map(BooleanParameter),
         "quote" -> cmd.csvQuote.map(StringParameter),
-        "escape" -> cmd.csvEscape.map(StringParameter)
+        "escape" -> cmd.csvEscape.map(StringParameter),
+        // increase the default limit on the number of columns if needed
+        // default is set at org.apache.spark.sql.execution.datasources.csv.CSVOptions maxColumns
+        "maxColumns" -> Option(LongParameter(max(20480, knownNumberOfFields)))
```

Comment:
Maybe set this option only if the limit is breached? We can put a lower limit in case it is changed in the future.
```scala
"maxColumns" -> if (knownNumberOfFields > 10000) Some(knownNumberOfFields) else None
```

Reply:
Thank you. Indeed, it is better not to change the defaults unless needed. I would like to compare to 20480 for the same reason.